lib/ansible/executor/module_common.py

```diff
@@ -65,6 +66,8 @@ def _find_snippet_imports(module_data, module_path, strip_comments):
     module_style = 'old'
     if REPLACER in module_data:
         module_style = 'new'
+    elif REPLACER_WINDOWS in module_data:
+        module_style = 'new'
```

Comment:
Changing this to a new style module is awesome because it enables check mode on Windows modules... However, we don't yet have a corresponding Windows mechanism for modules that aren't check mode-aware to be opted-out of running in check mode. A quick n' dirty solution might be to add another comment directive like # CHECK_MODE_AWARE...